extensions/wikidata/src/org/openrefine/wikidata/schema/WbDateConstant.java

```diff
@@ -104,9 +104,15 @@ public static TimeValue parse(String datestamp)
         Date bestDate = null;
         int precision = 0; // default precision (will be overridden if successfully parsed)
         int maxLength = 0; // the maximum length parsed
+        boolean bceFlag = false; // judge whether this is a BCE year
         String calendarIri = TimeValue.CM_GREGORIAN_PRO; // Gregorian calendar is assumed by default
         
         String trimmedDatestamp = datestamp.trim();
+
+        if(trimmedDatestamp.startsWith("-")){
+            trimmedDatestamp = trimmedDatestamp.replace("-", "");
+            bceFlag = true;
+        }
         
         if("TODAY".equals(trimmedDatestamp)) {
```

Comment:
Since you added your handling of the leading minus sign before the handling of `TODAY`, it means that we would parse `-TODAY` as some BCE date, which does not seem ideal. How about moving your BCE-handling block after the `TODAY` one? In this way, `-TODAY` would be rejected (perhaps it is worth adding a test about that).

Reply:
Yes, that's sure